src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,147 @@
+package tasks;
+
+/**
+ * The Task class represents a task in the TaskList.
+ */
+public class Task {
+
+    private String desc;
+    private boolean isDone;
+    private TaskType type;
+    private String start;
+    private String end;
+
+    /**
+     * The TaskType enum represents the type of task.
+     */
+    public enum TaskType {
+        TODO('T'), DEADLINE('D'), EVENT('E');
+
+        private char code;
+
+        TaskType(char code) {
+            this.code = code;
+        }
+
+        public char getCode() {
+            return code;
+        }
+    }
+
+    /**
+     * Constructor for the todo Task.
+     * @param desc
+     * @param isDone
+     */
+    public Task(String desc, boolean isDone) {
+        this.desc = desc;
+        this.isDone = isDone;
+        this.type = TaskType.TODO;
+    }
+
+    /**
+     * Constructor for the deadline task.
+     * @param desc
+     * @param isDone
+     * @param start
+     */
+    public Task(String desc, boolean isDone, String start) {
+        this.desc = desc;
+        this.isDone = isDone;
+        this.type = TaskType.DEADLINE;
+        this.start = start;
+    }
+
+    /**
+     * Constructor for the event Task
+     * @param desc
+     * @param isDone
+     * @param start
+     * @param end
+     */
+    public Task(String desc, boolean isDone, String start, String end) {
+        this.desc = desc;
+        this.isDone = isDone;
+        this.type = TaskType.EVENT;
+        this.start = start;
+        this.end = end;
+    }
+
+
+    /**
+    * Returns the task description.
+    * @return the task description
+    */
+    public String getDesc() {
+        return desc;
+    }
+
+    /**
+     * Returns true if the task is done, false otherwise.
+     * @return true if the task is done, false otherwise
+     */
+    public boolean isDone() {
+        return isDone;
+    }
+
+    /**
+     * Returns the type of the task.
+     * @return the type of the task
+     */
+    public TaskType getType() {
```

Comment:
Noted on nice way in implementing getter and setter!

Reply:
Thanks!